argmin/src/solver/particleswarm/mod.rs

```diff
@@ -50,7 +51,7 @@ use serde::{Deserialize, Serialize};
 /// \[1\] <https://en.wikipedia.org/wiki/Particle_swarm_optimization>
 #[derive(Clone)]
 #[cfg_attr(feature = "serde1", derive(Serialize, Deserialize))]
-pub struct ParticleSwarm<P, F> {
+pub struct ParticleSwarm<P, F, G> {
```

Comment:
Throughout argmin, `G` indicates a gradient, therefore using `G` for a generator may be confusing. I'm not sure though what a good alternative may be. Maybe `R` or even `Rng`?